src/dev/impl/DevToys/Helpers/CertificateHelper.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+
+namespace DevToys.Helpers
+{
+#nullable enable
+    internal static class CertificateHelper
+    {
+        private const string BeginCertificate = "BEGIN CERTIFICATE";
+        private const string BeginCertificateRequest = "BEGIN CERTIFICATE REQUEST";
+        internal const string IncorrectPassword = "The specified network password is not correct";
+        internal const string FormatNotSupported = "Certificate Request format not currently supported.";
+
+        /// <summary>
+        /// Returns the friendly formatted, decoded certificate details.
+        /// CSR format is currently not supported in UWP, but is in .NET 7 <see href="https://learn.microsoft.com/en-us/dotnet/api/system.security.cryptography.x509certificates.certificaterequest.loadsigningrequest?view=net-7.0"/>
+        /// PEM format is limited by the current framework version <see href="https://learn.microsoft.com/en-us/dotnet/api/system.security.cryptography.x509certificates.x509certificate2.createfrompemfile?view=net-7.0"/>
+        /// </summary>
+        /// <param name="input">Certificate data</param>
+        /// <param name="password">Password for protected private key.  This should only be relevant for decoding pfx format.</param>
+        /// <returns>Certificate details</returns>
+        internal static bool TryDecodeCertificate(string input, string? password, out string? decoded)
+        {
+            decoded = null;
+            string publicCert = input;
+            if (publicCert.Contains('-'))
+            {
+                string[] splitCert = input.Split('-', StringSplitOptions.RemoveEmptyEntries);
+                for (int i = 0; i < splitCert.Length; i++)
+                {
+                    if (splitCert[i] == BeginCertificate)
+                    {
+                        publicCert = splitCert[i + 1];
+                        break;
+                    }
+                    else if (splitCert[i] == BeginCertificateRequest)
+                    {
+                        decoded = FormatNotSupported;
+                        return false;
+                    }
+                }
+            }
+
+            try
+            {
+                var certificate = new X509Certificate2(Convert.FromBase64String(publicCert), password);
+                decoded = certificate.ToString();
+            }
+            catch (CryptographicException wce)
+            {
+                // If this is a valid certificate, but an incorrect/missing password, we should still return true
+                if (wce.Message == IncorrectPassword)
```

Comment:
I'm concerned that if Windows is in a different language, the message will be in a different language too and this condition won't work then.

Couple of suggestions:
1. By any chance, is there any other information we can get from the exception to identify this specific error? Maybe the [HResult](https://learn.microsoft.com/en-us/dotnet/api/system.exception.hresult?view=net-7.0#system-exception-hresult) property and detect `ERROR_INVALID_PASSWORD`?
2. Find the translation of this message in .NET / Windows (if any) and localize the message in DevToys so we can detect it depending on the language?

Reply:
Ah, great call! It looks like the `HResult` will work for this.  I tried various scenarios:
```csharp
using System.Security.Cryptography;
using System.Security.Cryptography.X509Certificates;

PrintHResult(() => {
    var input = Convert.FromBase64String("YXNkZmxrYWpzZGxrZmpha2xzZGZqbGthc2pkZmFzZGY=");
    var invalidCert = new X509Certificate2(input);
});

PrintHResult(() => new X509Certificate2(@"C:\certs\cert-does-not-exist.pfx"));

PrintHResult(() => new X509Certificate2(@"C:\certs\requires-certificate.pfx"));

void PrintHResult(Action action) {
    try {
        action();
    }
    catch (CryptographicException wce) {
        Console.WriteLine(wce.HResult);
    }
}
```

And seems to yield distinct output: 
```bash
>  dotnet run
-2146885623
-2147024894
-2147024810
```

Thank you - will follow up with a commit on this PR.